sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -288,6 +288,31 @@ void guessLocalWorkSize(size_t *threadsPerBlock, const size_t *global_work_size,
   }
 }
 
+pi_result enqueueEventsWait(pi_queue command_queue, CUstream stream,
+                            pi_uint32 num_events_in_wait_list,
+                            const pi_event *event_wait_list) {
+  try {
+    ScopedContext active(command_queue->get_context());
+
+    if (event_wait_list) {
```

Comment:
It may be faster to move the check to early in the function to exit early. Avoids the potential overhead of `ScopedContext` when there is no operation needed.

Reply:
done